### PR TITLE
feat: Pixelate an image

### PR DESCRIPTION
**Pull request by viveknathani**
Pixelating an image involves making square sections in your image
that have all their pixel values set to a common value. In our case,
we set all the values in a section/submatrix/window to the average
pixel value in that area.

### CODE REVIEW

**Review by priyanshul-govil — CHANGES_REQUESTED** [review]
1. `pixelRatio` does not seem a good name for the purpose it serves. *pixelWidth*, *windowWidth* might be clearer.
2. It's better to have more variables denoting *totalRed*, *totalGreen*, *totalBlue* as lines 482-484 don't make logical sense on their own. It'll be better to calculate lines 490-492 as `avgRed = totalRed / count` and so on.

**Comment by viveknathani** [comment]
Done

File: src/secur3dit/filters/Filters.java [hunk 1]
```diff
@@ -463,41 +463,41 @@ public static BufferedImage pixelate(BufferedImage image, int pixelRatio) {
         int avgGreen = 0;
```
Comment (priyanshul-govil) [inline src/secur3dit/filters/Filters.java]: variables should be declared in the scope they are used.

**Review by priyanshul-govil — CHANGES_REQUESTED** [review]
*average* variables have been declared a scope above which they are meant to be used.

**Comment by viveknathani** [comment]
Took care of that. And also removed a needless check for indices. Details are mentioned in the commit body.

**Review by priyanshul-govil — APPROVED** [review]
seems good now.

**Pull request merged**